qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -56,31 +56,66 @@ class ActiveSpaceTransformer(BaseTransformer):
+              1. the length of `active_orbitals` must be equal to `num_molecular_orbitals`.
               2. the sum of electrons present in `active_orbitals` must be equal to `num_electrons`.
 
     References:
         - *M. Rossmannek, P. Barkoutsos, P. Ollitrault, and I. Tavernelli, arXiv:2009.01872
           (2020).*
     """
 
-    def __init__(self, num_electrons: int, num_orbitals: int, num_alpha: Optional[int] = None,
-                 active_orbitals: Optional[List[int]] = None):
+    def __init__(self,
+                 num_electrons: Optional[int] = None,
+                 num_molecular_orbitals: Optional[int] = None,
+                 num_alpha: Optional[int] = None,
+                 active_orbitals: Optional[List[int]] = None,
+                 freeze_core: bool = False,
+                 remove_orbitals: Optional[List[int]] = None,
+                 ):
         """Initializes a transformer which can reduce a `QMolecule` to a configured active space.
 
+        This transformer requires the AO-basis matrices `hcore` and `eri` to be available, as well
+        as the basis-transformation matrix `mo_coeff`. A `QMolecule` produced by Qiskit's drivers in
+        general satisfies these conditions unless it was read from an FCIDump file. However, those
+        integrals are likely already reduced by the code which produced the file or can be
+        transformed using this driver after copying the MO-basis integrals of the produced
+        `QMolecule` into the AO-basis containers and initializing `mo_coeff` with an identity matrix
+        of appropriate size.
+
         Args:
-            num_electrons: the number of active electrons.
-            num_orbitals: the number of active orbitals.
-            num_alpha: the optional number of active alpha-spin electrons.
+            num_electrons: The number of active electrons. This may only be omitted if `freeze_core`
+                           is used.
+            num_molecular_orbitals: The number of active orbitals. This may only be omitted if
+                                    `freeze_core` is used.
+            num_alpha: The optional number of active alpha-spin electrons.
             active_orbitals: A list of indices specifying the molecular orbitals of the active
                              space. This argument must match with the remaining arguments and should
                              only be used to enforce an active space that is not chosen purely
                              around the Fermi level.
+            freeze_core: A convenience argument to quickly enable the inactivity of the
+                         `QMolecule.core_orbitals`. This keyword overwrites the use of all other
+                         keywords (except `remove_orbitals`) and, thus, cannot be used in
+                         combination with them.
+            remove_orbitals: A list of indices specifying molecular orbitals which are removed in
```

Comment:
More and more these feel like independent transformers. A subclass of this just having freeze_core and remove_orbitals that computes the active space, and internally delegates the actual transformation to an instance of this it creates, when it needs to do the transform, does that not work and make things simpler for an end user? I know having a separate class was discussed before but it was avoided due to complexity. Maybe I am missing something...

Reply:
Yes I was starting to think this as well... I will have a look tomorrow :+1: